iree/compiler/Dialect/Flow/IR/FlowOps.cpp

```diff
@@ -277,27 +279,55 @@ static LogicalResult verifyVariableStoreIndirectOp(
 // flow.dispatch.region
 //===----------------------------------------------------------------------===//
 
+/// Inlines operation |op| into the |dispatchRegionOp| by making all operands,
+/// as well as values caputred implicitly by the regions of the operation, that
+/// are outside the dispatch region operands of the dispatch region as well. For
+/// ConstantOp of scalar values or splat constants, the defining op is inlined
+/// as well.
+static Operation *inlineOpIntoDispatchRegion(OpBuilder &builder,
+                                             DispatchRegionOp dispatchRegionOp,
+                                             Operation *op,
+                                             BlockAndValueMapping &map) {
+  llvm::SetVector<Value> capturedInputs(op->getOperands().begin(),
+                                        op->getOperands().end());
+  getUsedValuesDefinedAbove(op->getRegions(), capturedInputs);
+  Block *block = builder.getInsertionBlock();
+  for (Value capturedInput : capturedInputs) {
+    if (map.contains(capturedInput)) continue;
+    // If the capturedInput is a constant scalar or splat constant clone it in
+    // the dispatch region.
+    if (ConstantOp constOp = capturedInput.getDefiningOp<ConstantOp>()) {
```

Comment:
Can you rely on RematerializeDispatchConstants instead? That way we aren't duplicating this logic with different behavior in multiple places.

Reply:
I knew there was a place this happened, but forgot where it was. I did this here because I ended up getting a compiler error when not doing that. I'll look into that. Agree about not duplicating the logic.

Reply:
duplication may still be worth it if it makes IR dumps easier to work through - just wanted to see if there was a reason (like maybe a fix needed for the pass etc) :)

Reply:
Yup, fix was needed for the pass. Its a simple fix. I just didnt know which pass did that.